nomos-services/key-management-system/src/keys/ed25519.rs

```diff
@@ -1,22 +1,49 @@
-#![allow(dead_code, reason = "KMS will be integrated soon.")]
-
 use bytes::Bytes;
 use ed25519_dalek::ed25519::signature::Signer as _;
-use overwatch::DynError;
 use serde::{Deserialize, Serialize};
 use zeroize::ZeroizeOnDrop;
 
-use crate::secure_key::SecuredKey;
+use crate::{
+    encodings::Encoding,
+    keys::{KeyError, secured_key::SecuredKey},
+};
+
+type InnerEncoding = crate::encodings::Bytes;
 
 #[derive(Serialize, Deserialize, ZeroizeOnDrop)]
 pub struct Ed25519Key(pub(crate) ed25519_dalek::SigningKey);
 
-impl SecuredKey for Ed25519Key {
-    fn sign(&self, data: Bytes) -> Result<Bytes, DynError> {
-        Ok(Bytes::copy_from_slice(&self.0.sign(&data).to_bytes()))
+impl SecuredKey<InnerEncoding> for Ed25519Key {
+    type Signature = InnerEncoding;
+    type PublicKey = InnerEncoding;
+    type Error = KeyError;
```

Comment:
@strinnityk [I meant something like](https://github.com/logos-co/nomos/pull/1708/files#r2376090300):

```rust
impl SecuredKey<InnerEncoding> for Ed25519Key {
    type Signature = [u8; 32];
    type PublicKey = [u8; 32];
    type Error = KeyError;
...
```
Or even the actual types. What does `Ed25519` use for pk's? Probably it has an specific type right?